### PR TITLE
Code refactoring and some other changes

### PR DESCRIPTION
**Pull request by Fireboyd78**
Some of the following changes have been made:
- Added `MemberFunc` to AGEHook sub-types
- Added `sdlPage16` class implementation
- Added `gfxTexture` class implementation (partial)
- Moved hooks to appropriate file for better organization
- Moved some hooks into the appropriate classes
- Removed Herobrine

### CODE REVIEW

File: src/mm2_gfx.h [hunk 1]
```diff
@@ -87,4 +87,101 @@ namespace MM2
...
+
+    class gfxTexture {
+    public:
+        /*0x00*/uint VglBindIndex;
+
+        /*0x04*/const char *Name;
+
+        /*0x06*/ushort Width;
+        /*0x08*/ushort Height;
+
```
Comment (0x1F9F1) [inline src/mm2_gfx.h:99]: I believe the offsets are meant to be 0x08 and 0x0A

Comment (Fireboyd78) [reply src/mm2_gfx.h:99]: Ahhhh fooey...I did it again! Damnit, lol.

Comment (Fireboyd78) [reply src/mm2_gfx.h:99]: Gonna fix this here shortly!

File: src/main.cpp [hunk 2]
```diff
@@ -1100,6 +984,8 @@ class vglHandler {
     static unsigned int GetAdjustedColor(gfxDrawMode drawMode, unsigned int color) {
         if (sdlPage16Handler::blockPtr != NULL)
         {
+            sizeof(gfxTexture);
```
Comment (0x1F9F1) [inline src/main.cpp]: I guess this isn't meant to be here?

Comment (Fireboyd78) [reply src/main.cpp]: Dear lord...

**Pull request merged**